### PR TITLE
Feature: last overflow from typeof this

### PR DESCRIPTION
**Pull request by sunrabbit123**
**Description:**
```ts
function Test3(this: { no: number }) {
  let x: typeof this.no = 1;
}
```

### CODE REVIEW

File: crates/stc_ts_file_analyzer/src/analyzer/stmt/var_decl.rs [hunk 1]
```diff
@@ -262,33 +262,68 @@ impl Analyzer<'_, '_> {
...
+                                ty.normalize(),
                                 Type::Query(QueryType {
-                                    span,
-                                    expr: box QueryExpr::TsEntityName(RTsEntityName::Ident(RIdent::new(
-                                        "globalThis".into(),
-                                        span.with_ctxt(SyntaxContext::empty()),
-                                    ))),
-                                    metadata: Default::default(),
-                                    tracker: Default::default(),
+                                    expr: box QueryExpr::TsEntityName(RTsEntityName::TsQualifiedName(box RTsQualifiedName {
```
Comment (kdy1) [inline crates/stc_ts_file_analyzer/src/analyzer/stmt/var_decl.rs:269]: I think this branch is reachable using other types than `globalThis`

Comment (kdy1) [reply crates/stc_ts_file_analyzer/src/analyzer/stmt/var_decl.rs:269]: Please use `ty.is_global_this()` instead.
https://rustdoc.stc.dudy.dev/stc_ts_types/enum.Type.html#method.is_global_this

**Review by kdy1 — CHANGES_REQUESTED** [review]
Thank you!

File: crates/stc_ts_file_analyzer/src/analyzer/stmt/var_decl.rs (hunk 1 above)
Comment (sunrabbit123) [reply crates/stc_ts_file_analyzer/src/analyzer/stmt/var_decl.rs:269]: this code mean `typeof this`

Comment (sunrabbit123) [reply crates/stc_ts_file_analyzer/src/analyzer/stmt/var_decl.rs:269]: just contains globalThis, this mean never only globalThis

Comment (sunrabbit123) [reply crates/stc_ts_file_analyzer/src/analyzer/stmt/var_decl.rs:269]: Do symbols also use query types?

Comment (sunrabbit123) [reply crates/stc_ts_file_analyzer/src/analyzer/stmt/var_decl.rs:269]: I'll check it out tomorrow.

File: crates/stc_ts_file_analyzer/src/analyzer/stmt/var_decl.rs [hunk 2]
```diff
@@ -262,33 +262,68 @@ impl Analyzer<'_, '_> {
...
+                                        left: RTsEntityName::Ident(..),
```
Comment (kdy1) [inline crates/stc_ts_file_analyzer/src/analyzer/stmt/var_decl.rs:270]: `Ident(..)` => `Ident(RIdent { sym: js_word!("this"), .. })`

**Review by kdy1 — APPROVED** [review]
Thanks!

**Review by kdy1 — CHANGES_REQUESTED** [review]
Can you rebase? I tried rebasing but seems like there's a conflict

**Review by kdy1 — APPROVED** [review]
Thanks!

**Pull request merged**